src/core/Sampler/Sampler.cpp

```diff
@@ -233,6 +236,229 @@ void Sampler::noteOff(Note* pNote )
 }
 
 
+// functions for pan parameters and laws-----------------
+
+float Sampler::getRatioPan( float fPan_L, float fPan_R ) {
+   /** returns the single pan parameter in [-1,1] from the L,R gains
+	* as it was input from the GUI (up to scale and translation, which is arbitrary)
+	*/
+	if ( fPan_L < 0. || fPan_R < 0. || ( fPan_L == 0. && fPan_R == 0.) ) { // invalid input
+		WARNINGLOG( "Invalid (panL, panR): both zero or some is negative. Pan set to center." );
+		return 0.; // default central value
+	} else {
+		if ( fPan_L >= fPan_R ) {
+			return fPan_R / fPan_L - 1.;
+		} else {
+			return 1. - fPan_L / fPan_R;
+		}
+	}
+}
+
+   /** PAN LAWS
```

Comment:
Nice. Thanks for the thorough documentation! Could you move it above the `enum` definition in the header? Right now, it will be shown as the description of the ` ratioStraightPolygonalPanLaw()` function.